tools/codeowners-utils/Azure.Sdk.Tools.CodeownersUtils/Parsing/CodeownersEntry.cs

```diff
@@ -192,5 +192,135 @@ int AddHashCodeForObject(int hc, object item)
+                    .ToList();
+
+                var uniqueSourceOwners = new List<string>();
+                var seenSource = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var owner in normalizedSourceOwners)
+                {
+                    if (seenSource.Add(owner))
+                    {
+                        uniqueSourceOwners.Add("@" + owner);
+                    }
+                }
+
+                // Compute pad width based on the path length (start at basePad and increase in steps of 5)
+                int padWidth = ComputePad(path.Length);
+                var pathLine = $"{path}".PadRight(padWidth) + $"{string.Join(" ", uniqueSourceOwners)}";
+                lines.Add(pathLine);
+            }
+
+            if (addSeparationLine)
+            {
+                lines.Add("");
+            }
+
+            // Add ServiceLabel(s) on a single line if provided
+            if (serviceLabels.Any())
+            {
+                var formattedServiceLabels = serviceLabels
+                    .Where(lbl => !string.IsNullOrWhiteSpace(lbl))
+                    .Select(lbl => lbl.StartsWith("%") ? lbl : $"%{lbl}");
+                lines.Add($"# ServiceLabel: {string.Join(" ", formattedServiceLabels)}");
+            }
+
+            // Add AzureSDKOwners if provided (normalize/dedupe)
+            if (azureSDKOwners != null && azureSDKOwners.Count > 0)
+            {
+                var normalizedAzureSdkOwners = azureSDKOwners
+                    .Where(o => !string.IsNullOrWhiteSpace(o))
+                    .Select(o => o.Trim())
+                    .Select(o => o.TrimStart('@'))
+                    .Select(o => o.Trim())
+                    .ToList();
+
+                var uniqueAzureSdkOwnersList = new List<string>();
+                var seenAzure = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var owner in normalizedAzureSdkOwners)
+                {
+                    if (seenAzure.Add(owner))
+                    {
+                        uniqueAzureSdkOwnersList.Add("@" + owner);
+                    }
+                }
+
+                // Use computed pad width so owners align even when the path is long
+                int padForAzure = ComputePad(Math.Max(path.Length, "# AzureSdkOwners: ".Length));
+                var azureSDKOwnersLine = $"# AzureSdkOwners: ".PadRight(padForAzure) + $"{string.Join(" ", uniqueAzureSdkOwnersList)}";
+                lines.Add(azureSDKOwnersLine);
+            }
+
+            // Add ServiceOwners if provided (normalize/dedupe)
+            if (serviceOwners != null && serviceOwners.Count > 0)
+            {
+                var normalizedServiceOwners = serviceOwners
+                    .Where(o => !string.IsNullOrWhiteSpace(o))
+                    .Select(o => o.Trim())
+                    .Select(o => o.TrimStart('@'))
+                    .Select(o => o.Trim())
```

Comment:
```suggestion
                    .Select(o => o.Trim().TrimStart('@').Trim())
```

chained `.Select`s can be simplified to a single select. I think the latter is more readable but it's a matter of preference.